wallet-test/src/test/scala/org/bitcoins/wallet/WalletSendingTest.scala

```diff
@@ -210,6 +211,51 @@ class WalletSendingTest extends BitcoinSWalletTest {
     }
   }
 
+  it should "correctly send entire outpoints" in { fundedWallet =>
+    val wallet = fundedWallet.wallet
+    for {
+      allUtxos <- wallet.listUtxos()
+      // use half of them
+      utxos = allUtxos.drop(allUtxos.size / 2)
+      outPoints = utxos.map(_.outPoint)
+      tx <- wallet.sendFromOutPoints(outPoints, testAddress, None)
+    } yield {
+      val expectedFeeRate =
+        wallet.feeRateApi.asInstanceOf[RandomFeeProvider].lastFeeRate.get
+      assert(outPoints.forall(outPoint =>
+               tx.inputs.exists(_.previousOutput == outPoint)),
+             "Every outpoint was not included included")
+      assert(tx.inputs.size == outPoints.size, "An extra input was added")
+
+      val inputAmount = utxos.map(_.output.value).sum
+      val numInputs = outPoints.size
+
+      val defaultRange = expectedFeeRate.scaleFactor * numInputs
+
+      val (actualFeeRate, range) = expectedFeeRate match {
+        case _: SatoshisPerByte =>
+          (SatoshisPerByte.calc(inputAmount, tx), defaultRange)
+        case _: SatoshisPerVirtualByte =>
+          (SatoshisPerVirtualByte.calc(inputAmount, tx), defaultRange)
+        case _: SatoshisPerKiloByte =>
+          (SatoshisPerKiloByte.calc(inputAmount, tx), defaultRange)
+        case _: SatoshisPerKW =>
+          // multiply range by 4 because an extra byte on a sig counts as 4 weight units
+          (SatoshisPerKW.calc(inputAmount, tx), defaultRange * 4)
+      }
+
+      // +- range in case of rounding or unexpected signature sizes
+      assert(
+        actualFeeRate.toLong === expectedFeeRate.toLong +- range,
+        s"Expected fee rate: $expectedFeeRate, inputs: $numInputs input: $inputAmount, tx bytes: ${tx.byteSize} " +
+          s"vsize: ${tx.vsize} weight ${tx.weight}"
+      )
+
+      assert(tx.outputs.size == 1)
+      assert(tx.outputs.head.scriptPubKey == testAddress.scriptPubKey)
```

Comment:
Probably need an assertion on the amount in the output too?

Reply:
It is by the fee rate assertion